### PR TITLE
Logging to sentry

### PR DESCRIPTION
**Pull request by relekang**
- [x] Add reporting of failed services #47 
- [x] Add sentry logging handler
- [x] Add `--loglevel` option

### CODE REVIEW

File: frigg_worker/cli.py [hunk 1]
```diff
@@ -1,20 +1,14 @@
 # -*- coding: utf8 -*-
+import logging
 import logging.config
```
Comment (frecar) [inline frigg_worker/cli.py]: do we need to import logging and logging.config?

File: frigg_worker/cli.py [hunk 2]
```diff
@@ -23,17 +17,36 @@ def load_logging_config():
...
+        logger.info('Starting frigg worker')
+        fetcher(**options)
     except Exception as e:
-        print(e)
         logger.error(e)
         sentry.captureException()
 
 
+def evaluate_options(options):
+    if options['dispatcher_url'] is None:
```
Comment (frecar) [inline frigg_worker/cli.py]: The four ``if options['something'] is None:`` may be generalized into:

    def set_option(key, options, config):
        options[key] = config[key.upper]

and then replaced by 

    set_option(hq_token, options, config)
    set_option(hq_url, options, config)
    set_option(dispatcher_url, options, config)
    etc...

File: frigg_worker/cli.py (hunk 1 above)
Comment (relekang) [inline frigg_worker/cli.py]: I'm not sure I added it based on a import error.

File: frigg_worker/jobs.py [hunk 3]
```diff
@@ -118,7 +118,7 @@ def run_tests(self):
             self.finished = True
             self.report_run()
 
-            logger.info("Run of build %s finished." % self.id)
+            logger.info('Run of build %s finished.' % self.id)
```
Comment (frecar) [inline frigg_worker/jobs.py]: Should we be consistent with use of % or {} ?

File: frigg_worker/jobs.py [hunk 4]
```diff
@@ -130,23 +130,24 @@ def clone_repo(self, depth=1):
...
-            message = "Access denied to %s/%s" % (self.owner, self.name)
+            message = 'Access denied to %s/%s' % (self.owner, self.name)
             logger.error(message)
         return clone.succeeded
 
     def start_services(self):
         for service in self.settings['services']:
-            self.docker.run("sudo service {0} start".format(service))
+            if not self.docker.run('sudo service {0} start'.format(service)).succeeded:
+                logger.warning('Service "{}" did not start.')
```
Comment (frecar) [inline frigg_worker/jobs.py]: You will need to format the log message with the correct service name?

File: tests/test_cli.py [hunk 5]
```diff
@@ -17,13 +15,7 @@ def test_start(self, mock_fetcher):
...
         self.assertTrue('Starting frigg worker' in result.output)
         self.assertEqual(result.exit_code, 0)
-        mock_fetcher.assert_called_once_with(
-            dispatcher_token=None,
-            dispatcher_url=None,
-            hq_token=None,
-            hq_url=None,
-            slack_url=None
-        )
+        mock_fetcher.assert_called_once()
```
Comment (frecar) [inline tests/test_cli.py]: It this done because you are lazy, or because you don't know the parameters? ^^

File: tests/test_fetcher.py [hunk 6]
```diff
@@ -29,30 +22,15 @@ def test_start_build(self, mock_runtests, mock_docker_stop, mock_docker_start):
...
         self.assertTrue(mock_docker_stop.called)
 
-    @patch('frigg_worker.fetcher.config', mock_config)
     @responses.activate
     def test_fetch_task(self):
 
         responses.add(
             responses.GET,
-            mock_config('DISPATCHER_URL'),
+            'http://example.com',
```
Comment (frecar) [inline tests/test_fetcher.py]: hard coded?

File: frigg_worker/cli.py (hunk 2 above)
Comment (relekang) [inline frigg_worker/cli.py]: The function is just moved, so it outside the scope of this change. However I don't see how the changes are only added if the option is None?

File: tests/test_jobs.py [hunk 7]
```diff
@@ -201,9 +201,13 @@ def test_start_four_services_in_order(self, mock_docker_run):
 
         mock_docker_run.assert_has_calls([
             mock.call('sudo service redis-server start'),
+            mock.call().succeeded.__bool__(),
```
Comment (frecar) [inline tests/test_jobs.py]: what do you test here?

File: frigg_worker/cli.py (hunk 2 above)
Comment (frecar) [inline frigg_worker/cli.py]: You could add an if to the set_option.. But anyway, we can refactor this later.

File: frigg_worker/jobs.py (hunk 3 above)
Comment (relekang) [inline frigg_worker/jobs.py]: Yes :+1: 

I think it deserves a separate PR

File: tests/test_fetcher.py (hunk 6 above)
Comment (relekang) [inline tests/test_fetcher.py]: Yes, read line 35, it is hard coded there as well. It is like hard-coding in `assertEquals`

File: tests/test_jobs.py (hunk 7 above)
Comment (relekang) [inline tests/test_jobs.py]: Since the mock returns a MagickMock all calls to any attribute of that object will be in the call list, thus if statement in [`if not self.docker.run('sudo service {0} start'.format(service)).succeeded:`](https://github.com/frigg/frigg-worker/pull/50/files#diff-259a6cde6c4c0cbdb0dd42c579112a1eR149) will result in `mock.call().succeeded.__bool__()`

File: frigg_worker/jobs.py (hunk 4 above)
Comment (relekang) [inline frigg_worker/jobs.py]: Oops, should be fixed now :)

File: frigg_worker/cli.py (hunk 1 above)
Comment (relekang) [inline frigg_worker/cli.py]: Removed in b9fa784

File: frigg_worker/cli.py (hunk 2 above)
Comment (relekang) [inline frigg_worker/cli.py]: It would be nice if we somehow could use `options.update` :)

**Comment by frecar** [comment]
:shipit
